src/compounds/navigation-dropdown/src/index.tsx

```diff
@@ -25,10 +26,21 @@ const NavigationDropdown: React.FC<DropdownProps> = ({
 
   const handleClick = () => setOpen(!open)
   const handleItemClick = (e: any) => {
-    setSelection(e?.target?.innerText || selection)
+    const childContent = e?.target?.children
+    if (
+      childContent?.length &&
+      childContent[0]?.innerText &&
+      Object.values(navConfig).includes(childContent[0]?.innerText)
+    ) {
+      if (
+        childContent[0]?.innerText === 'Please select' ||
```

Comment:
I thought this comes from the schema

Reply:
It accepts rich text block and lists, headings, buttons, and so on inside that rich text block. Since I haven't cloned or created new React element which enables me to pass click handlers, I need to understand which link is clicked in order to set dropdown's value.

Reply:
If the clicked link is Please select or the current page, it will prevent the default event and won't reload the same page